langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/SchemaHelper.java

```diff
@@ -0,0 +1,96 @@
+package dev.langchain4j.model.vertexai;
+
+import com.google.cloud.vertexai.api.Schema;
+import com.google.cloud.vertexai.api.Type;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Helper class to create a <code>com.google.cloud.vertexai.api.Schema</code>
+ * from a JSON schema string, or from a class by reflection on its public fields.
+ */
+public class SchemaHelper {
+
+    /**
+     * Create an instance of <code>Schema</code> from a JSON schema string.
+     * @param jsonSchemaString the JSON schema string
+     * @return a fully built schema
+     */
+    public static Schema fromJsonSchema(String jsonSchemaString) {
+        try {
+            // ensure types are in uppercase
+            // Open API JSON Schema normally use lowercase types,
+            // but it seems the Gemini SDK mandates uppercase types
+            // as exposed in <code>com.google.cloud.vertexai.api.Type</code>
+            String schemaJsonString = jsonSchemaString
+                .replace("\"object\"", "\"OBJECT\"")
+                .replace("\"integer\"", "\"INTEGER\"")
+                .replace("\"string\"", "\"STRING\"")
+                .replace("\"number\"", "\"NUMBER\"")
+                .replace("\"array\"", "\"ARRAY\"")
+                .replace("\"boolean\"", "\"BOOLEAN\"");
+
+            Schema.Builder builder = Schema.newBuilder();
+            JsonFormat.parser().merge(schemaJsonString, builder);
+            return builder.build();
+        } catch (InvalidProtocolBufferException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Create an instance of <code>Schema</code> from a class by reflection on its fields.
+     * @param theClass the class for which to create a schema representation
+     * @return a fully built schema
+     */
+    public static Schema fromClass(Class<?> theClass) {
```

Comment:
It seems that this kind of logic is already implemented in `ToolSpecifications`, maybe worth looking into re-using it.